internal/impl/kafka/output_sarama_kafka.go

```diff
@@ -68,6 +69,8 @@ Unfortunately this error message will appear for a wide range of connection prob
 			docs.FieldString("key", "The key to publish messages with.").IsInterpolated(),
 			docs.FieldString("partitioner", "The partitioning algorithm to use.").HasOptions("fnv1a_hash", "murmur2_hash", "random", "round_robin", "manual"),
 			docs.FieldString("partition", "The manually-specified partition to publish messages to, relevant only when the field `partitioner` is set to `manual`. Must be able to parse as a 32-bit integer.").IsInterpolated().Advanced(),
+			docs.FieldInt("partitions_per_new_topic", "If a topic is created and this value is greater than zero then this number of partitions will be used for the topic.").Optional().Advanced(),
+			docs.FieldInt("topic_replication_factor", "The replication factor for the newly created topics. If `partitions_per_new_topic` is greater than zero this field is required, otherwise is ignored").Advanced(),
```

Comment:
Been thinking over the config both in terms of intuition when opting into manual topic creation and also the assumed behaviour when these fields aren't set. The current behaviour is that topic creation is left up to your broker configuration, where it may be enabled or not.

If a user wishes to opt-into this new functionality it's not entirely necessary for them to set both the partitions and the number of replicas as they may want to stick with the broker defaults, and this applies to either field.

With that in mind I think there are two issues:
1. The presence of these fields could be confusing as users may no longer assume that the broker defaults would be respected if they omit these fields
2. Using partitions_per_new_topic > 0 as the determining factor as to whether manual creation is unintuitive for users that want to use the broker default for partitions but a manual setting of replicas

I think a more intuitive approach would be to add a config block that is more explicit:

```
    custom_topic_creation:
      enabled: false
      partitions: -1
      replication_factor: -1
```

The name `custom_topic_creation` makes it clearer that the default is to defer to whatever the broker sets. Having an explicit `enabled` field removes any ambiguity around which fields are necessary for enabling this functionality, and setting defaults for both fields at -1 aligns it with other APIs in the kafka ecosystem and is a more explicit signal of "use the broker settings".

Reply:
I agree that a whole config object for this functionality, like the one you proposed, is definitely more intuitive for the user. The only thing is that unfortunately the function `sarama.ClusterAdmin.CreateTopic()` requires a `sarama.TopicDetail` object that has both `NumPartitions` and `ReplicationFactor` set. That is understandable, because the default value of one could cause conflicts with the new value of the other, for example setting the number of partitions to 2 for a new topic but having the replication factor set to 3 by default.

So I would implement the changes that you proposed, but still require both `partitions` and `replication_factor` to be set